treeherder/webapp/api/performance_data.py

```diff
@@ -743,6 +748,250 @@ def _filter_out_retriggers(serialized_data: List[dict]) -> List[dict]:
+            'platform__platform',
+            'test',
+            'option_collection_id',
+            'repository_id',
+            'measurement_unit',
+        )
+        new_signatures = new_signatures.values(
+            'framework_id',
+            'id',
+            'extra_options',
+            'suite',
+            'platform__platform',
+            'test',
+            'option_collection_id',
+            'repository_id',
+            'measurement_unit',
+        )
+
+        base_signature_ids = [item['id'] for item in list(base_signatures)]
+        new_signature_ids = [item['id'] for item in list(new_signatures)]
+
+        base_perf_data = PerformanceDatum.objects.select_related('push', 'repository', 'id').filter(
+            signature_id__in=base_signature_ids,
+            repository__name=base_repository_name,
+        )
+        new_perf_data = PerformanceDatum.objects.select_related('push', 'repository', 'id').filter(
+            signature_id__in=new_signature_ids,
+            repository__name=new_repository_name,
+        )
+
+        # TODO: Allow comparison when selecting only new revisions
+
+        if base_revision:
+            base_perf_data = base_perf_data.filter(push__revision=base_revision)
+        elif interval:
+            base_perf_data = base_perf_data.filter(
+                push_timestamp__gt=datetime.datetime.utcfromtimestamp(
+                    int(time.time() - int(interval))
+                )
+            )
+        if new_revision:
+            new_perf_data = new_perf_data.filter(push__revision=new_revision)
+        elif interval:
+            new_perf_data = new_perf_data.filter(
+                push_timestamp__gt=datetime.datetime.utcfromtimestamp(
+                    int(time.time() - int(interval))
+                )
+            )
+
+        option_collection_map = self.get_option_collection_map()
+
+        base_grouped_job_ids, base_grouped_values = self.get_grouped_perf_data(base_perf_data)
+        new_grouped_job_ids, new_grouped_values = self.get_grouped_perf_data(new_perf_data)
+
+        base_signatures_map, base_names, base_platforms = self.get_signatures_map(
+            base_signatures, base_grouped_values, option_collection_map
+        )
+        new_signatures_map, new_names, new_platforms = self.get_signatures_map(
+            new_signatures, new_grouped_values, option_collection_map
+        )
+
+        names = set(base_names + new_names)
+        platforms = set(base_platforms + new_platforms)
+        self.queryset = []
+
+        for header in names:
+            for platform in platforms:
+                key = '{} {}'.format(header, platform)
+                base_result = base_signatures_map.get(key, {})
+                new_result = new_signatures_map.get(key, {})
+                is_empty = False if base_result and new_result else True
+                if is_empty:
+                    continue
+                suite = base_result.get('suite', '')
```

Comment:
you can move these rows directly where you define `row_result`, for example `suite` and `test`, to avoid having variaables you don't use.